### PR TITLE
Jon's Dev Guide section

### LINKED ISSUE #22: As a developer, I can view use cases
... so I can learn from examples how users are meant to use the product.

### PR DESCRIPTION
**Pull request by fullfatwasabi**
closes #22

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -283,32 +283,65 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User chooses to edit the staff details.
+2.  User inputs relevant details.
+3.  Staff’d requests for confirmation.
+4.  User confirms.
+5. Staff’d updates the new staff details.
 
-    Use case ends.
+   Use case ends.
```
Comment (Gabau) [inline docs/DeveloperGuide.md:298]: Why is this indentation off?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -283,32 +283,65 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-**Use case: Delete a person**
+**Use case: UC01 - Edit staff details**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User chooses to edit the staff details.
```
Comment (Gabau) [inline docs/DeveloperGuide.md:292]: Can change all the numbers to 1 following GFMD standard.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -283,32 +283,65 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * a1. Staff’d requests to confirm the cancellation.
+    * a2. User confirms the cancellation.
 
-*{More to be added}*
+      Use case ends.
+
+**Use case: UC02 - Edit staff schedule**
+
+**MSS**
+
```
Comment (Gabau) [inline docs/DeveloperGuide.md:320]: Change numbers to 1 for this as well?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -3,7 +3,7 @@ layout: page
 title: Developer Guide
 ---
 * Table of Contents
-{:toc}
+  {:toc}
```
Comment (Gabau) [inline docs/DeveloperGuide.md:6]: Why the extra whitespace?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -283,32 +283,65 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User chooses to edit the staff details.
+2.  User inputs relevant details.
+3.  Staff’d requests for confirmation.
+4.  User confirms.
+5. Staff’d updates the new staff details.
```
Comment (Gabau) [inline docs/DeveloperGuide.md:296]: Inconsistent spacing.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -283,32 +283,65 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 3a2. User enters new data.
+    * 3a2. Steps 3a1-3a2 are repeated until the data entered are correct.
 
-* 3a. The given index is invalid.
+      Use case resumes at step 4.
 
-    * 3a1. AddressBook shows an error message.
 
-      Use case resumes at step 2.
+* *a At any time, User chooses to cancel the edit.
```
Comment (Gabau) [inline docs/DeveloperGuide.md:311]: How do we cancel the edit when the edit command states the edit? I do not think this extension is feasible. Unless you want to change how edit works?

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -283,32 +283,65 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 3a. Staff'd detects an error in the entered data.
+
+    * 3a1. Staff'd displays an error message.
+    * 3a2. User enters new data.
+    * 3a2. Steps 3a1-3a2 are repeated until the data entered are correct.
+
+      Use case resumes at step 4.
+
+
+* *a At any time, User chooses to cancel the edit.
```
Comment (Gabau) [inline docs/DeveloperGuide.md:340]: > How do we cancel the edit when the edit command states the edit? I do not think this extension is feasible. Unless you want to change how edit works?

Same for this. Maybe a redo command would be possible?

**Review by Gabau — CHANGES_REQUESTED** [review]
Some issues with indentation and conflicting opinion on the edit command. Don't really know how many use cases to put but be aware of the following. 

image

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (fullfatwasabi) [reply docs/DeveloperGuide.md:292]: Change all the numbers in the steps to 1?

Comment (Gabau) [reply docs/DeveloperGuide.md:292]: I was thinking of something like this.

image

Comment (fullfatwasabi) [reply docs/DeveloperGuide.md:292]: Hmm this part is the steps in the use case tho, I don't think it makes sense to have them all be 1 haha, anyway I went back to see the use case in ab3 and it was numbered similarly too. Unless I'm interpreting this wrongly and you mean something else?

Comment (Gabau) [reply docs/DeveloperGuide.md:292]: Should be fine. Just thought it might make it easier to edit the MSS later on.

Comment (mweeruien) [reply docs/DeveloperGuide.md:292]: I also believe it should be listed normally. The use case example given by AB3 isn't listed with all 1s. 
image

Comment (Gabau) [reply docs/DeveloperGuide.md:292]: GFMD formats all 1s in a list to ordered.

Comment (fullfatwasabi) [reply docs/DeveloperGuide.md:292]: Hmm ya but for the extension part like 3a1 it doesn't tho so I thought something more consistent would be better, we can do both tho I guess

Comment (fullfatwasabi) [reply docs/DeveloperGuide.md:292]: Okie nvm I have no strong feelings about this I guess I've updated it

File: docs/AboutUs.md [hunk 8]
```diff
@@ -31,9 +31,9 @@ You can reach us at the email `seer[at]comp.nus.edu.sg`
 
 ### Johnny Doe
```
Comment (Gabau) [inline docs/AboutUs.md:32]: Need to change the name to follow #21 except for the picture.

**Review by Gabau — CHANGES_REQUESTED** [review]
LGTM, but #24 overwrote #21. Have no idea how that happened.

File: docs/AboutUs.md (hunk 8 above)
Comment (fullfatwasabi) [reply docs/AboutUs.md:32]: yeah I've already updated the rest! :)

**Pull request merged**